synapse/lib/stormtypes.py

```diff
@@ -2371,6 +2372,14 @@ def __init__(self, runt, name, info):
         self.locls.update(self.getObjLocals())
         self.locls['name'] = self.name
 
+    def __hash__(self):
+        return hash(self.name)
+
+    def __eq__(self, othr):
```

Comment:
What is __eq__ used for?

Reply:
CPython calls `__eq__` in the case where it finds an existing with the same hash. Stuff like `$lib.queue.get` needs this in order to dedup